diff --git a/build/ffmpeg-wasm.sh b/build/ffmpeg-wasm.sh
index faa2f72..c789b7b 100755
--- a/build/ffmpeg-wasm.sh
+++ b/build/ffmpeg-wasm.sh
@@ -35,7 +35,7 @@ CONF_FLAGS=(
   -sUSE_SDL=2                              # use emscripten SDL2 lib port
   -sSTACK_SIZE=5MB                         # increase stack size to support libopus
   -sMODULARIZE                             # modularized to use as a library
-  ${FFMPEG_MT:+ -sINITIAL_MEMORY=1024MB}   # ALLOW_MEMORY_GROWTH is not recommended when using threads, thus we use a large initial memory
+  ${FFMPEG_MT:+ -sINITIAL_MEMORY=2048MB}   # ALLOW_MEMORY_GROWTH is not recommended when using threads, thus we use a large initial memory
   ${FFMPEG_MT:+ -sPTHREAD_POOL_SIZE=32}    # use 32 threads
   ${FFMPEG_ST:+ -sINITIAL_MEMORY=32MB -sALLOW_MEMORY_GROWTH} # Use just enough memory as memory usage can grow
   -sEXPORT_NAME="$EXPORT_NAME"             # required in browser env, so that user can access this module from window object
diff --git a/build/ffmpeg.sh b/build/ffmpeg.sh
index da89c45..71cd343 100755
--- a/build/ffmpeg.sh
+++ b/build/ffmpeg.sh
@@ -30,4 +30,4 @@ CONF_FLAGS=(
 )
 
 emconfigure ./configure "${CONF_FLAGS[@]}" $@
-emmake make -j
+emmake make -j4
